### ISSUE TITLE
[Bug?]: auth-provider header missing

### ISSUE THREAD

**Issue by gilliardmacedo** [redwoodjs/redwood#8367]
### What's not working?

I'm working on a dbAuth app, and making changes to our deployment. We currently use serverless framework.

After some changes, I started to get currentUser as null in the graphql function, even with auth function answering ok for any call.

Investigating the auth flow in the Redwood codebase, I saw this as part of nullability flow of current user:

[packages/api/src/auth/index.ts](https://github.com/redwoodjs/redwood/blob/main/packages/api/src/auth/index.ts#L9)
```
// This is shared by `@redwoodjs/web`
const AUTH_PROVIDER_HEADER = 'auth-provider'

export const getAuthProviderHeader = (event: APIGatewayProxyEvent) => {
  return event?.headers[AUTH_PROVIDER_HEADER]
}
```
(getting the header in lowercase)

And my browser:

![Screenshot 2023-05-18 at 23 23 28](https://github.com/redwoodjs/redwood/assets/26877564/a9554c7d-a0e1-48d6-963a-ccdc7ffff466)
(sending capitalized)

I don't know which codebase or infrastructure change affected the header case, but since headers are case insensitive, I think it's better to fix. The workaround currently is copy the header to the expected case.

It doesn't occurs locally, only at AWS.

### What's your environment? (If it applies)

```shell
System:
    OS: macOS 13.3.1
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.16.0 - /private/var/folders/m1/k1cyqcz57g5_60mpqtxrjxd40000gn/T/xfs-797e81c3/node
    Yarn: 3.5.1 - /private/var/folders/m1/k1cyqcz57g5_60mpqtxrjxd40000gn/T/xfs-797e81c3/yarn
  Databases:
    SQLite: 3.39.5 - /usr/bin/sqlite3
  Browsers:
    Chrome: 113.0.5672.126
    Firefox: 113.0.1
    Safari: 16.4
  npmPackages:
    @redwoodjs/auth-dbauth-setup: 5.2.1 => 5.2.1 
    @redwoodjs/core: 5.2.1 => 5.2.1
```

### Are you interested in working on this?

- [X] I'm interested in working on this

**Comment by jtoar** [comment]
Hey @gilliardmacedo, I don't see the harm in the solution you suggested, but just to be thorough, could we take one more step: in my devtools pane, the header shows up as capitalized too, but you have to hit the "raw" checkbox to see the real casing. Mind double checking just so we're sure that this is the issue?

<img width="1392" alt="Screenshot 2023-05-19 at 11 35 48 AM" src="https://github.com/redwoodjs/redwood/assets/32992335/08f8d055-228c-497e-9115-811980717a32">

**Comment by gilliardmacedo** [comment]
My Chrome doesn't have the checkbox:

![Screenshot 2023-05-19 at 17 26 25](https://github.com/redwoodjs/redwood/assets/26877564/d0f8e09d-76b5-4367-9e4e-dfa79fecdc5a)

When I try to copy to a cURL request, all headers are lowercase.

But I can see in Firefox in lowercase as you saw:

![Screenshot 2023-05-19 at 17 25 12](https://github.com/redwoodjs/redwood/assets/26877564/a04dc228-1376-4a40-bb9f-f025dcf9eb7e)

Since headers should be case-insensitive([ref](https://www.rfc-editor.org/rfc/rfc9110.html#name-field-names)), I think my Chrome version has a different behavior(or CloudFront doing something with AllViewerExceptHostHeader policy).

In the backend I could see the Auth-Provider with capital letters:

![Screenshot 2023-05-19 at 17 35 14](https://github.com/redwoodjs/redwood/assets/26877564/10ca4377-a566-4c5d-8a4a-eba62152caf1)

But authorization in lowercase:
![Screenshot 2023-05-19 at 17 37 05](https://github.com/redwoodjs/redwood/assets/26877564/fc50e4da-b04a-4169-bdd1-aa6ae232e319)

**Comment by jtoar** [comment]
@gilliardmacedo I merged your fix into the next branch and released it as an RC; could you try it and make sure it works?

```
yarn rw upgrade -t rc
```

The version should be: `5.2.2-rc.0+5afacb4af`. With or without the stuff after the "+". Here's the version on NPM: https://www.npmjs.com/package/@redwoodjs/core/v/5.2.2-rc.0

**Comment by gilliardmacedo** [comment]
Tested with `5.2.2-rc.0` and without the workaround and it's working. Thanks again, @jtoar

**Comment by jtoar** [comment]
@gilliardmacedo great news!

Heads up that I'm going to release a patch (v5.2.2), but it won't include this fix, so don't upgrade yet. The team is confused why the header's cased changed and wants to do more digging before we officially release a fix for it. We'd also appreciate any leads into what you think happened if anything—this was a sudden change based on deploy provider right?

**Comment by gilliardmacedo** [comment]
Yes. I was using AWS Amplify and moved to default serverless deployment. But different from default deployment, my API requests pass through CloudFront too to do reverse proxy. This API behavior uses AWS-provided AllViewerExceptHostHeader policy for headers, that means it'll forward everything except the host. I may be the reason of case change.

**Comment by thedavidprice** [comment]
Very helpful and interesting @gilliardmacedo As you know, we're having challenges with using Serverless Framework + Redwood. If you are available:
- could you take a look at our default serverless config for API and Web (here in the [Deploy Target CI project](https://github.com/redwoodjs/deploy-target-ci/tree/main/serverless-aws))
- and document the modifications you made in this [Issue about outstanding Serverless challenges](https://github.com/redwoodjs/deploy-target-ci/issues/826)

Ideally we wouldn't have to handle CORS with dbAuth, which seems it might be resolved by your settings/solution.

I still don't know why requests don't trigger if you go directly to a Post route (for example).

**Comment by gilliardmacedo** [comment]
Sure, @thedavidprice. The challenge now is to automate those changes I did after the serverless deployment. I was already planning to help Redwood community if I had success.

**Comment by dthyresson** [comment]
Should this issue be closed due to the fix in #8474?

**Comment by jtoar** [comment]
Thanks @dthyresson, yes that was the intention behind that PR, let's close this
